### PR TITLE
chore(common): CHECKOUT-6445 Move offline component to separate package

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
As above

## Why?
As part of decomposing payment integrations, move `<OfflinePaymentMethod/>` to it's own package and use the new Payment integration API.

## Testing / Proof
- Circle

@bigcommerce/checkout

### CODE REVIEW

File: packages/offline-payment-integration/src/OfflinePaymentMethod.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    });
+
+    it('catches error during ApplePay initialization', async () => {
+        jest.spyOn(checkoutService, 'initializePayment').mockRejectedValue(new Error('test error'));
+        mount(<OfflinePaymentMethod {...props} />);
+
+        await new Promise((resolve) => process.nextTick(resolve));
+
+        expect(props.onUnhandledError).toHaveBeenCalled();
+    });
```
Comment (bc-peng) [inline packages/offline-payment-integration/src/OfflinePaymentMethod.spec.tsx:61]: Is this related to ApplePay?

Comment (animesh1987) [reply packages/offline-payment-integration/src/OfflinePaymentMethod.spec.tsx]: copy paste 🤦

File: packages/offline-payment-integration/tsconfig.lib.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "include": [
+    "src/**/*.ts",
+    "src/**/*.tsx"
+  ]
+}
```
Comment (bc-peng) [inline packages/offline-payment-integration/tsconfig.lib.json:7]: I must miss this one before. Think we discussed removing `lib.json` before. 
The rest look good to me.

File: packages/offline-payment-integration/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "references": [
+    {
+      "path": "./tsconfig.lib.json"
```
Comment (bc-peng) [inline packages/offline-payment-integration/tsconfig.json:5]: Same here.

**Review by bc-peng — APPROVED** [review]

File: packages/offline-payment-integration/tsconfig.lib.json (hunk 2 above)
Comment (animesh1987) [reply packages/offline-payment-integration/tsconfig.lib.json:7]: I can't recall correctly the discussion, but can remove this since don't think it's used.
